lib/beacon/config.ex

```diff
@@ -422,36 +422,55 @@ defmodule Beacon.Config do
     "#{cat}/*"
   end
 
-  defp validate_allowed_media_types!(allowed_media_types) do
-    Enum.each(allowed_media_types, fn media_type ->
-      case Plug.Conn.Utils.media_type(media_type) do
-        {:ok, _, "*", _} ->
+  defp validate_allowed_media_accept_types!(allowed_media_accept_types) do
+    Enum.each(allowed_media_accept_types, fn media_type ->
+      validate_media_accept_type(media_type)
+    end)
+  end
+
+  defp validate_media_accept_type(media_type) do
+    if String.contains?(media_type, "/") do
+      do_validate_media_accept_type(media_type)
+    else
+      validate_accept_extension(media_type)
+    end
+  end
+
+  defp do_validate_media_accept_type(media_type) do
+    case Plug.Conn.Utils.media_type(media_type) do
+      {:ok, category, "*", _} ->
+        if Enum.member?(["image", "audio", "video"], category) do
+          media_type
+        else
           raise Beacon.LoaderError, """
-          Catchall Media Types are not allowed in allowed
+          Catchall Media Types are only allowed for `image`, `audio`, `video` media types.
+          Media Type: #{media_type}
           """
+        end
 
-        :error ->
-          raise_invalid_media_type(media_type)
+      :error ->
+        raise_invalid_media_type(media_type)
 
-        _ ->
-          media_type
-      end
-    end)
+      _ ->
+        media_type
+    end
   end
 
-  defp process_assets_config(allowed_media_types, assigned_assets) do
+  # .some_ext (not checking for valid extension)
+  defp validate_accept_extension(<<46, _rest::binary>> = media_type), do: media_type
+
+  # catchall
+  defp validate_accept_extension(media_type), do: raise("`#{media_type}` does not appear to be a media type, extensions must begin with a `.`")
```

Comment:
You most likely want to raise LoaderError here

```suggestion
  defp validate_accept_extension(media_type), do: raise Beacon.LoaderError, "`#{media_type}` does not appear to be a media type, extensions must begin with a `.`"
```

Reply:
yes!